pkg/proxy/iptables/proxier.go

```diff
@@ -664,30 +673,33 @@ func updateServiceMap(
 
 func (proxier *Proxier) OnEndpointsAdd(endpoints *api.Endpoints) {
 	namespacedName := types.NamespacedName{Namespace: endpoints.Namespace, Name: endpoints.Name}
-	proxier.endpointsChanges.update(&namespacedName, nil, endpoints)
-
-	proxier.syncProxyRules(syncReasonEndpoints)
+	if proxier.endpointsChanges.update(&namespacedName, nil, endpoints) {
+		// TODO: Check if endpoints and services are synced.
```

Comment:
Same answer as above - clarified the TODO. And will send a separate PR to fix it (it doesn't matter until we don't have periodic runner).